docs/source/tutorials/datasets.rst

```diff
@@ -108,33 +153,60 @@ Here is an example of sample that is formatted with :class:`~torchtune.data.Alpa
 
 We provide `other instruct templates <https://github.com/pytorch/torchtune/blob/main/torchtune/data/_instruct_templates.py>`_
 for common tasks such summarization and grammar correction. If you need to create your own
-instruct template for a custom task, you can create your own :class:`~torchtune.data.InstructTemplate`
-class and point to it in the config.
+instruct template for a custom task, you can inherit from :class:`~torchtune.data.InstructTemplate`
+and create your own class.
+
+.. code-block:: python
+
+    from torchtune.datasets import instruct_dataset
+    from torchtune.data import InstructTemplate
+
+    class CustomTemplate(InstructTemplate):
+        # Define the template as string with {} as placeholders for data columns
+        template = ...
+
+        # Implement this method
+        @classmethod
+        def format(
+            cls, sample: Mapping[str, Any], column_map: Optional[Dict[str, str]] = None
+        ) -> str:
+            ...
+
+    # Load in tokenizer
+    tokenizer = ...
+    dataset = instruct_dataset(
+        tokenizer=tokenizer,
+        source="my/dataset/path",
+        template="import.path.to.CustomTemplate",
+    )
 
 .. code-block:: yaml
 
+    # YAML config
     dataset:
       _component_: torchtune.datasets.instruct_dataset
-      source: mydataset/onthehub
-      template: CustomTemplate
-      train_on_input: True
-      max_seq_len: 512
+      source: my/dataset/path
+      template: import.path.to.CustomTemplate
+
+.. code-block:: bash
+
+    # Command line
+    tune run full_finetune_single_device --config llama3/8B_full_single_device dataset=torchtune.datasets.instruct_dataset dataset.source=my/dataset/path dataset.template=import.path.to.CustomTemplate
+
+
+Custom chat dataset and chat formats
```

Comment:
I think the breakdown here between instruct and chat is good for now. But some of this information, in the future should be contained in the Chat/Instruct api reference and this tutorial should provide the extra high level information (like how to customize things) about all our dataset formats. This is assuming that the number of formats grows much more.

Reply:
Yeah make sense, I would say the level of detail here is sufficient for tutorial / general usage but anything more should go in the API reference. lmk if I should generalize this a bit more